src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -176,15 +178,14 @@ bool ZProbe::run_probe(int& steps, bool fast)
 
     // Enable the motors
     THEKERNEL->stepper->turn_enable_pins_on();
-    this->current_feedrate = (fast ? this->fast_feedrate : this->slow_feedrate) * Z_STEPS_PER_MM; // steps/sec
-    float maxz= this->max_z*2;
```

Comment:
yea i would suggest a probe.max_travel (or max_z) but then the code needs to allow for legacy configs that do not have that specified and default to exactly what it was before. the rule of least surprises to legacy users is needed here.
and as I said each of these changes needs its own PR, it makes it easier to review and see what else may break. Pay particular attention to the levelling strategies that rely on zprobe functionality to not change.